Source/JavaScriptCore/API/JSValue.h

```diff
@@ -549,11 +549,7 @@ Create a JSValue from a CGRect.
  */
 @interface JSValue (PropertyAccess)
 
-#if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED < 101500) || (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED < 130000)
```

Comment:
This is API, and people are presumably still able to set deployment target to before 10.15? If so, this change seems incorrect.

Reply:
Can you explain how that makes sense? Are we ensuring all our new APIs work with those older macOS versions? And given the code we're removing elsewhere you'd presumably run into build problems too.

Reply:
When a customer installs newest Xcode with newest SDK, they are still able to select an older deployment target, if they want their software to work on older OS versions. These`..._MIN_REQUIRED` macros are part of what changes then.

So this is not about building WebKit, but about building 3rd party software that uses WebKit. The system framework doesn't change on old OSes, so we need to keep our latest SDK compatible with old WebKit.

I don't know if Xcode even has any limit on how far back one can go. Of course there are some natural boundaries (e.g. one cannot use a deployment target that only supported 32-bit Intel, because Xcode has dropped support for 32-bit Intel).

Now, I'm not sure about what exactly is going on here. It is possible that this code is incorrect, and doesn't actually achieve the goal. You can either revert this particular change for now, or investigate deeper.

Reply:
> Are we ensuring all our new APIs work with those older macOS versions?

I don't think we have any automation which ensures that API changes are backwards-compatible. We have availability annotations to prevent developers from calling methods that may not exist on their deployment target. And we tend to avoid ABI-sensitive declarations in public headers (i.e. no C++, no C structs). But in general we rely on extra reviewer scrutiny on public headers.

> And given the code we're removing elsewhere you'd presumably run into build problems too.

Well, this is the only public header in the PR. So we can safely remove the other code here, because we don't support building _WebKit itself_ for these older OS versions. But we do support 3rd-party developers using a modern SDK to target very old OS versions.

So I think you need to undo the changes to this file, and let the rest of the PR stand.